eternalcore-api/src/main/java/com/eternalcode/core/feature/warp/WarpService.java

```diff
@@ -7,15 +7,21 @@
 
 public interface WarpService {
 
-    Warp createWarp(String name, Location location);
+    Warp create(String name, Location location);
 
-    void removeWarp(String warp);
+    void delete(String warp);
 
-    boolean warpExists(String name);
+    void addPermissions(String warp, String... permissions);
+
+    void removePermission(String warp, String permission);
+
+    boolean exists(String name);
+
+    boolean hasPermission(String warp, String permission);
 
     Optional<Warp> findWarp(String name);
 
-    Collection<String> getNamesOfWarps();
+    Collection<String> getAllNames();
 
-    boolean hasWarps();
+    boolean isEmpty();
```

Comment:
ja bym to w sumie wywalił te isEmpty() i getAllNames() i dodać getWarps() czy coś takiego, będzie wtedy pełne API